src/Microsoft.ML.Core/Data/MetadataUtils.cs

```diff
@@ -336,6 +336,13 @@ public static bool HasKeyNames(this Schema schema, int col, int keyCount)
                 && type.ItemType.IsText;
         }
 
+        public static bool HasKeyNames(this SchemaShape.Column col)
```

Comment:
>HasKeyNames [](start = 27, length = 11)

`HasKeyValues` then, right? Also, I would be cautious about making it public. You could always to with 'internal-bestfriend' I guess